src/components/Editor/Preview/Popup.js

```diff
@@ -44,24 +53,37 @@ export class Popup extends Component<Props> {
   marker: any;
   pos: any;
 
-  componentDidMount() {
+  async componentWillMount() {
     const {
-      loadObjectProperties,
-      loadedObjects,
       value,
-      editor,
-      range
+      expression,
+      setPopupObjectProperties,
+      popupObjectProperties
     } = this.props;
+    const root = createNode(null, expression, expression, { value });
+
+    if (
+      !nodeIsPrimitive(root) &&
+      value &&
+      value.actor &&
+      !popupObjectProperties[value.actor]
+    ) {
+      const onLoadItemProperties = loadItemProperties(root, createObjectClient);
+      if (onLoadItemProperties !== null) {
+        const properties = await onLoadItemProperties;
+        setPopupObjectProperties(value, properties);
+      }
+    }
```

Comment:
this looks like it could be moved to a util in the debugger or reps project.

I'm thinking this because it pulls in a bunch of dependencies and is pretty complicated...

could we move some of this logic to the `actions/pause/setupPopup...`